pkg/natdiscovery/natdiscovery.go

```diff
@@ -129,7 +140,7 @@ func (nd *natDiscovery) AddEndpoint(endpoint *types.SubmarinerEndpoint) {
 	remoteNAT := newRemoteEndpointNAT(endpoint)
 
 	// support nat discovery disabled or a remote cluster endpoint which still hasn't implemented this protocol
-	if _, err := extractNATDiscoveryPort(endpoint); err == errorNoNatDiscoveryPort || nd.serverPort == nil {
+	if _, err := extractNATDiscoveryPort(endpoint); err == errorNoNatDiscoveryPort || nd.serverPort == 0 {
```

Comment:
We probably shouldn't ignore a parse error even tho it **should** never occur. 

```suggestion
	if _, err := extractNATDiscoveryPort(endpoint); err != nil || nd.serverPort == 0 {
	    if err != errorNoNatDiscoveryPort {
	        klog.Errorf("Error extracting NATT discovery port from endpoint %q: %v", endpoint.Spec.CableName, err)
	    }
```

Reply:
Done.